content/chainguard/migration/porting-apps-to-chainguard/index.md

```diff
@@ -17,62 +17,35 @@ weight: 20
 
+* Chainguard's distroless Images have no shell or package manager by default. This is great for security, but sometimes you need these things, especially in builder images. For those cases we have `-dev` images (such as `cgr.dev/chainguard/python:latest-dev`) which do include a shell and package manager.
+* Chainguard Images typically don't run as root, so a `USER root` statement may be required before installing software.
+* The `-dev` images and `wolfi-base` / `chainguard-base` use BusyBox by default, so any `groupadd` or `useradd` commands will need to be ported to `addgroup` and `adduser`.
+* The free Developer tier of Images provides `:latest` and `:latest-dev` versions. Our paid Production Images offer tags for major and minor versions.
+* We use apk tooling, so `apt install` commands will become `apk add`.
+* Chainguard Images are based on `glibc` and our packages cannot be mixed with Alpine packages.
+* In some cases, the entrypoint in Chainguard Images can be different from equivalent images based on other distros, which can lead to unexpected behavior. You should always check the image's specific documentation to understand how the entrypoint works.
+* When needed, Chainguard recommends using a base image like `chainguard-base` or a `-dev` image to install an application's OS-level dependencies.
+* Although `-dev` images are still more secure than most popular container images based on other distros, for increased security on production environments we recommend combining them with a distroless variant in a multi-stage build.
 ---
 
 ## The Sample Application
 
-The application in question is [identidock](https://github.com/using-docker/identidock). This
-application was written for the book [Using
-Docker](https://learning.oreilly.com/library/view/using-docker/9781491915752/) about ten years ago,
-which shows that we can still migrate software of this age to a new container while realizing the
-benefits of a no-to-low CVE count. The application itself will create
-[identicons](https://en.wikipedia.org/wiki/Identicon) for a user name, similar to what [GitHub
-generates for users with no avatar](https://github.blog/2013-08-14-identicons/). It was designed at
-the time to demonstrate a "microservices" approach, and as such it's made up of 3 services:
-
+The application in question is [identidock](https://github.com/using-docker/identidock). This application was written for the book [Using Docker](https://learning.oreilly.com/library/view/using-docker/9781491915752/) about ten years ago, which shows that we can still migrate software of this age to a new container while realizing the benefits of a no-to-low CVE count. The application itself will create [identicons](https://en.wikipedia.org/wiki/Identicon) for a user name, similar to what [GitHub generates for users with no avatar](https://github.blog/2013-08-14-identicons/). It was designed at the time to demonstrate a "microservices" approach, and as such it's made up of 3 services:
```

Comment:
I think the manual breaks turn to be a hassle depending on your editor and resolution, so it feels like an accessibility issue to me - I do not intend to go around changing all articles, but when editing an existing article this way, it ends up causing spacing issues and such. I believe most markdown editors have a "Soft Wrap" option to visually break down the lines, I thought it was a more common feature but maybe I'm mistaken... agree it's probably not worth go around fixing this but if if causes accessibility issues it might be worth doing it while you're making changes.